src/ui_framework/README.md

```diff
@@ -7,9 +7,42 @@
 
 ## What is this?
 
-The UI Framework provides you with UI components you can quickly use to build UIs, as well as interactive examples which document how they're supposed to be used. These UI components are currently only implemented in CSS and markup, but eventually they'll grow to involve JS as well.
+The Kibana UI Framework provides you with UI components you can quickly use to build user interfaces for Kibana.
 
-When you build a UI using this framework (e.g. a plugin's UI), you can rest assured it will fit into the overall Kibana UI.
+The UI Framework comes with interactive examples which document how to use its various UI components. These components are currently only implemented in CSS, but eventually they'll grow to involve JS as well.
+
+When you build a UI using this framework (e.g. a plugin's UI), you can rest assured it will integrate seamlessly into the overall Kibana UI.
+
+## How to create a new component
+
+There are two steps to creating a new component:
+
+1. Create the CSS for the component in `ui_framework/components`.
+2. Document it with examples in `ui_framework/doc_site`.
+
+### Create the component CSS
+
+1. Create a directory for your component in `ui_framework/components`.
+2. Create an `_index.scss` file in this directory.
+3. Create at least one component SCSS file in the same directory.
```

Comment:
Maybe some guidance about the naming, like below. `_{component name}.scss`. Changing the order might be a little clearer, and remove some of the text as well.

1. Create `_{component name}.scss`
2. [Optional] Create any other logical grouping components
3. Create an `_index.scss` that imports all of the new component scss files you created